shell/platform/android/io/flutter/embedding/engine/plugins/activity/ActivityControlSurface.java

```diff
@@ -48,9 +54,31 @@
    * Dart code, the {@link Activity} should invoke this method. At that point the {@link
    * FlutterEngine} is considered "attached" to the {@link Activity} and all {@link ActivityAware}
    * plugins are given access to the {@link Activity}.
+   *
+   * @deprecated Prefer using the {@link #attachToActivity(ExclusiveAppComponent, Lifecycle)} API to
+   *     avoid situations where multiple activities are driving the FlutterEngine simultaneously.
+   *     See https://github.com/flutter/flutter/issues/66192.
    */
+  @Deprecated
```

Comment:
I am surprised this is public API?

Reply:
I don't have any proof that anyone's using it but it is public API. The theoretical usage I think would be fore someone who make a custom activity that holds a FlutterView. This public interface is the APIs that lets that FlutterView's owner tell that FlutterView's FlutterEngine's plugins when an activity is available / unavailable. 

That's why there's that activity/exclusiveActivity thing above. It's just transitional so we have a not-too-rough breaking change.

Reply:
That's fair.

So, there will be an announcement and a deprecation phase so we can clean it up?

Reply:
Yes, writing website PR at the moment.